solr/core/src/test/org/apache/solr/search/TestSolrQueryParser.java

```diff
@@ -1057,7 +1057,35 @@ public void testShingleQueries() throws Exception {
         , "/response/numFound==1"
     );
   }
-  
+
+
+  public void testSynonymQueryStyle() throws Exception {
+    ModifiableSolrParams edismaxParams = params("qf", "t_pick_best_foo");
```

Comment:
Just a minor point here but you needn't have a SolrParams based variable; you could simply inline it at each invocation.  This makes it easier to read each test request.  If you were trying to share some common params across test invocations then I could understand.